File: qoobot-service/community/qoocommunity-cloud/qoocommunity-event/src/main/java/com/qoobot/qoocommunity/event/domain/Event.java
package com.qoobot.qoocommunity.event.domain;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "events")
@EqualsAndHashCode(callSuper = true)
public class Event extends com.qoobot.qoocommunity.common.entity.BaseEntity {

    @Column(nullable = false, length = 500)
    private String title;

    @Column(nullable = false, unique = true, length = 200)
    private String slug;

    @Column(nullable = false, length = 30)
    private String type;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "content_html", columnDefinition = "TEXT")
    private String contentHtml;

    @Column(name = "cover_url", length = 512)
    private String coverUrl;

    @Column(length = 500)
    private String location;

    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;

    @Column(name = "end_time", nullable = false)
    private LocalDateTime endTime;

    @Column(length = 50)
    private String timezone = "Asia/Shanghai";

    @Column(name = "max_attendees")
    private Integer maxAttendees;

    @Column(name = "current_attendees")
    private Integer currentAttendees = 0;

    @Column(length = 20)
    private String status = "DRAFT";

    @Column(name = "is_featured")
    private Boolean isFeatured = false;

    @Column(name = "created_by", nullable = false, length = 64)
    private String createdBy;
}

File: qoobot-web/community/src/api/event.ts
import client from './client'

export interface Event {
  id: number
  title: string
  slug: string
  type: string
  description: string
  coverUrl: string
  location: string
  startTime: string
  endTime: string
  maxAttendees: number
  currentAttendees: number
  status: string
  isFeatured: boolean
}

export const eventApi = {
  getEvents: (params?: { type?: string; status?: string }) => client.get('/v1/events', { params }),
  getEvent: (id: number) => client.get(`/v1/events/${id}`),
  register: (eventId: number, data: { name?: string; company?: string; title?: string; email?: string }) =>
    client.post(`/v1/events/${eventId}/register`, data),
  cancelRegistration: (eventId: number) => client.delete(`/v1/events/${eventId}/register`),
  getAttendees: (eventId: number) => client.get(`/v1/events/${eventId}/attendees`),
  getMaterials: (eventId: number) => client.get(`/v1/events/${eventId}/materials`),
  getCalendar: () => client.get('/v1/events/calendar')
}
